File: 125. Valid Palindrome/solution.java
package com.company;

public class Main {

    public static void main(String[] args) {
        String s="race a car";
        String s1="A man, a plan, a canal: Panama";
        boolean a= isPalindrome(s);
       boolean b= isPalindrome(s1);
       System.out.println(a);
        System.out.println(b);
    }

    public static boolean isPalindrome(String s) {
        String _s=filter(s);
        char[] chars=_s.toCharArray();
        for (int i=0;i<chars.length-1-i;i++){
            if (chars[i]!=chars[chars.length-1-i]){
                return false;
            }
        }
        return true;
    }
    private  static  String filter(String s){
        char[] chars=s.trim().toLowerCase().toCharArray();
        StringBuffer s1=new StringBuffer();
        for (int i=0;i<chars.length;i++){
            if (Character.isAlphabetic(chars[i]) || Character.isDigit(chars[i])){
                s1.append(chars[i]);
            }
        }
        return s1.toString();
    }
}

File: 125. Valid Palindrome/solution.cs
using System.Collections.Generic;
using System.Linq;
using System.Collections;
using System;
using System.Text;
using System.Net.Http;

namespace a
{
	public class Solution
	{
		public static void Main()
		{	
			var s = "A man, a plan, a canal: Panama";
			Console.WriteLine(IsPalindrome(s));
			Console.ReadLine();
		}
		public  static bool IsPalindrome(string s)
		{
			if (s == "") {
				return true;
			}
			s = s.ToLower();
			int i=0;
			int l = s.Length;
			int j = l - 1;
			while (i < j) {
				if (char.IsLetterOrDigit(s[i])) {
					if (char.IsLetterOrDigit(s[j])) {
						if (s[i] == s[j]) {
							i++;
							j--;
						} else {
							return false;
						}
							
					} else {
						j--;
					}
				} else {
					i++;
				}
			} 
			return true;
		}
				
				

	}
	   
}


